sympy/solvers/tests/test_solvers.py

```diff
@@ -2290,3 +2290,13 @@ def test_issue_19509():
         -d + a + sqrt(-b + c),
         d + a - sqrt(-b - c),
         d + a + sqrt(-b - c)]
+
+def test_issue_20747():
+    THT, HT, DBH, dib, c0, c1, c2, c3, c4  = symbols('THT HT DBH dib c0 c1 c2 c3 c4')
+    f = c2 + (c3*DBH) + (c4 * THT)
+    rhs = 1 - (((HT-1)/(THT - 1))**c1)* (1- exp(c0/f))
+    assert solve(dib - DBH * (c0 - f * log(rhs)), HT) == [THT * ((1
+        - exp((DBH * c0 - dib) / (DBH * (DBH * c3 + THT * c4 + c2))))
+        / (1 - exp(c0 / (DBH * c3 + THT * c4 + c2)))) ** (1 / c1) - ((1
+        - exp((DBH * c0 - dib) / (DBH * (DBH * c3 + THT * c4 + c2))))
+        / (1 - exp(c0 / (DBH * c3 + THT * c4 + c2)))) ** (1 / c1) + 1]
```

Comment:
I think it would be better to write this as:
```
eq = ...
sol = ...
assert solve(eq, HT) == sol
```
Also if we can format the solutions a bit better that would be good. Try to use spacing around operators to help show what the expression is e.g. `a*x + b`. The spacing should be the same on both sides of an operator. Brackets are not needed in e.g. `x + (y*z)`.

Reply:
Okay i will make it like this. Thanks for reviewing

Reply:
Hey does this look good.
```
  THT, HT, DBH, dib, c0, c1, c2, c3, c4  = symbols('THT HT DBH dib c0 c1 c2 c3 c4')
  f = DBH*c3 + THT*c4 + c2
  rhs = 1 - ((HT - 1)/(THT - 1))**c1*(1 - exp(c0/f))
  eqn = dib - DBH*(c0 - f*log(rhs))
  sol = [THT*((1 - exp((DBH*c0 - dib)/(DBH*(DBH*c3 + THT*c4 + c2))))
         /(1 - exp(c0/(DBH*c3 + THT*c4 + c2))))**(1/c1)
         - ((1 - exp((DBH*c0 - dib)/(DBH*(DBH*c3+ THT*c4 + c2))))
         /(1 - exp(c0/(DBH*c3 + THT*c4+ c2))))**(1/c1) + 1]
  assert solve(eqn, HT) == sol
```

Reply:
I'd factor out the common term like:
```python
  THT, HT, DBH, dib, c0, c1, c2, c3, c4  = symbols('THT HT DBH dib c0 c1 c2 c3 c4')
  f = DBH*c3 + THT*c4 + c2
  rhs = 1 - ((HT - 1)/(THT - 1))**c1*(1 - exp(c0/f))
  eq = dib - DBH*(c0 - f*log(rhs))
  term = ((1 - exp((DBH*c0 - dib)/(DBH*(DBH*c3 + THT*c4 + c2))))
            / (1 - exp(c0/(DBH*c3 + THT*c4 + c2))))
  sol = [THT*term**(1/c1) - term**(1/c1) + 1]
  assert solve(eq, HT) == sol
```